driver-mapping/src/main/java/com/datastax/driver/mapping/CommonNamingConventions.java

```diff
@@ -0,0 +1,358 @@
+     * and words are not separated by any special character. E.g. "myxmlparser".
+     */
+    public static class LowerCase implements NamingConvention {
+
+        @Override
+        public List<Word> split(String input) {
+            List<Word> result = new ArrayList<Word>(1);
+            result.add(new Word(input));
+            return result;
+        }
+
+        @Override
+        public String join(List<Word> input) {
+            StringBuilder builder = new StringBuilder();
+            for (Word word : input) {
+                builder.append(word.getValue());
+            }
+            return builder.toString().toLowerCase();
+        }
+
+        @Override
+        public boolean equals(NamingConvention other) {
+            return other != null && other instanceof LowerCase;
+        }
+
+    }
+
+    // Abstract
+
+    abstract public static class CamelCase implements NamingConvention {
+
+        private static final String SPLIT_PATTERN = String.format("%s|%s|%s",
+                "(?<=[A-Z])(?=[A-Z][a-z])",
+                "(?<=[^A-Z])(?=[A-Z])",
+                "(?<=[A-Za-z])(?=[^A-Za-z])"
+        );
+
+        private static final String ALL_UPPERCASE_PATTERN = "([A-Z])*";
+
+        private static final Comparator<String> STRING_LENGTH_COMPARATOR = new Comparator<String>() {
+            @Override
+            public int compare(String o1, String o2) {
+                return o2.length() - o1.length();
+            }
+        };
+
+        private final String ignorablePrefixPattern;
+
+        protected CamelCase(String... ignorablePrefixes) {
+            Arrays.sort(ignorablePrefixes, STRING_LENGTH_COMPARATOR);
+            StringBuilder builder = new StringBuilder();
+            for (int i = 0; i < ignorablePrefixes.length; i++) {
+                if (i > 0) {
+                    builder.append("|");
+                }
+                builder.append("^");
+                builder.append(ignorablePrefixes[i]);
+            }
+            ignorablePrefixPattern = builder.toString();
+        }
+
+        @Override
+        public List<Word> split(String input) {
+            List<Word> result = new LinkedList<Word>();
+            // slice all ignorable prefixes, then split
+            for (String value : input.replaceAll(ignorablePrefixPattern, "").split(SPLIT_PATTERN)) {
+                // if all uppercase, mark as abbreviation (e.g. MyXMLParser)
+                boolean isAbbreviation = value.matches(ALL_UPPERCASE_PATTERN);
+                Word word = new Word(value, isAbbreviation);
+                result.add(word);
+            }
+            return result;
+        }
+
+        @Override
+        public boolean equals(NamingConvention other) {
+            return other instanceof CamelCase &&
+                    ((CamelCase) other).ignorablePrefixPattern.equals(ignorablePrefixPattern);
+        }
+
+    }
+
+    abstract public static class CharDelimitedNamingConvention implements NamingConvention {
```

Comment:
I don't think we should make this abstract since none of its members are abstract and might be useful to create anonymous naming conventions (although not sure why anyone would do that :D).